QemuPkg/FrontPageButtons/FrontPageButtons.c

```diff
@@ -0,0 +1,230 @@
+/** @file FrontpageButtons.c
+
+ This module installs the MsButtonServicesProtocol and reports the requested state of Vol+ and
+ Vol- as indicated by the smbios table type3 record version string.
+
+ If the Type3.version string is:
+     "Vol+"  -- Return Vol+ pressed.
+     "Vol-"  -- Return Vol- pressed.
+     any other string, or no string, return neither pressed
+
+  Use only for Qemu platforms which can change smbios every time it boots.
+
+  Copyright (C) Microsoft Corporation. All rights reserved.
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+
+**/
+#include <Uefi.h>
+
+#include <Protocol/ButtonServices.h>
+#include <Protocol/Smbios.h>
+
+#include <Library/BaseLib.h>
+#include <Library/DebugLib.h>
+#include <Library/MemoryAllocationLib.h>
+#include <Library/UefiBootServicesTableLib.h>
+
+typedef enum {
+  NoButtons     = 0,
+  VolUpButton   = 1,
+  VolDownButton = 2
+} BUTTON_STATE;
+
+#define SMBIOS_VOLUP    "Vol+"
+#define SMBIOS_VOLDOWN  "Vol-"
+
+BUTTON_STATE  ButtonState = NoButtons;
+
+/*
+Say volume button is pressed because we wan to go to frontpage.

```

Comment:
nit: `want` not `wan`

Also, for a non-static function, can we add @param/@retval comments here, same for below functions?

Reply:
Fixed